zospy/analyses/polarization.py

```diff
@@ -0,0 +1,344 @@
+import os
+import re
+import struct
+
+from tempfile import mkstemp
+from io import StringIO
+
+import numpy as np
+import pandas as pd
+
+from zospy import utils
+from zospy.analyses.base import AnalysisResult, AttrDict, process_oncomplete
+from zospy.api import constants
+import zospy.api.config as _config
+
+
+def polarization_pupil_map(oss, jx=1, jy=0, x_phase=0, y_phase=0, wavelength=1, field=1,

```

Comment:
Add type hints and use the `zospy.analyses.base.OnComplete` Enum for `OnComplete` (see other analyses for example usages)